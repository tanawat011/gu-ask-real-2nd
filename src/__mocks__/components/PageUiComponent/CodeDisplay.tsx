// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck

export const CodeDisplay = ({ code }) => {
  return <div data-testid='code-display'>{code}</div>
}
