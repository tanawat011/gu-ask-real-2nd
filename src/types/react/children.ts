export type Children = { children: JSX.Element }
