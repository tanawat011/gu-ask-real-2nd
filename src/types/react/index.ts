export * from './children'
export * from './reactRouterDom'
