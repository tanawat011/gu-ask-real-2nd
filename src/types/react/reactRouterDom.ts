import type { Scroll } from 'types/common'

export type OutletContextProps = {
  scroll: Scroll
}
