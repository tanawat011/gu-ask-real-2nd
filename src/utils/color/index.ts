export * from './hex2Rgb'
export * from './rgb2Hex'
