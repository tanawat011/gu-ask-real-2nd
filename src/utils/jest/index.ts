export * from './renderWithProviders'
export * from './twColor'
