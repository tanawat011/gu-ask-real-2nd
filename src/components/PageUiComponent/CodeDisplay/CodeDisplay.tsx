import { useState } from 'react'

import { PrismAsyncLight as SyntaxHighlighter } from 'react-syntax-highlighter'
import { oneDark as theme } from 'react-syntax-highlighter/dist/esm/styles/prism'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { IconCode, IconCopy } from 'components/Icons'

type CodeDisplayProps = {
  code: string
}

const TwContainer = tw.div`rounded-b-lg border-independence border-t bg-charcoal px-5 py-2`
const TwWrapButton = tw.div`flex justify-end`
const TwWrapButton2 = tw.div`grid grid-flow-col gap-1`

export const CodeDisplay: React.FC<CodeDisplayProps> = ({ code }) => {
  const [isShowCode, setIsShowCode] = useState(false)

  const handleShowCode = () => {
    setIsShowCode(!isShowCode)
  }

  return (
    <TwContainer>
      <TwWrapButton>
        <TwWrapButton2>
          <Button variant='plain' size='xs' icon={<IconCopy />} />

          <Button variant='plain' size='xs' icon={<IconCode />} onClick={handleShowCode} />
        </TwWrapButton2>
      </TwWrapButton>

      {isShowCode && (
        <SyntaxHighlighter language='tsx' style={theme}>
          {code}
        </SyntaxHighlighter>
      )}
    </TwContainer>
  )
}
