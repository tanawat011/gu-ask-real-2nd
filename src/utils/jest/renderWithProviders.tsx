import type { RenderResult } from '@testing-library/react'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { RecoilRoot } from 'recoil'

const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  const queryClient = new QueryClient()

  return (
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <BrowserRouter>{children}</BrowserRouter>
      </RecoilRoot>
    </QueryClientProvider>
  )
}

interface RenderWithProvidersOptions {
  router?: {
    route: string
  }
  recoil?: object
}

export const renderWithProviders = (
  ui: JSX.Element,
  option?: RenderWithProvidersOptions,
): RenderResult => {
  const routerOption = { ...(option?.router || { route: '/' }) }
  window.history.pushState({}, 'Test page', routerOption.route)

  return {
    ...render(ui, { wrapper: AllTheProviders, ...routerOption }),
  }
}
