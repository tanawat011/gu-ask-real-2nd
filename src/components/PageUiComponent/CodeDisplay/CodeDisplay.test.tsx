import { fireEvent, screen } from '@testing-library/react'

import { renderWithProviders } from 'utils/jest'

import { CodeDisplay } from './CodeDisplay'

jest.mock('@fortawesome/react-fontawesome')
jest.mock('react-syntax-highlighter')
jest.mock('react-syntax-highlighter/dist/esm/styles/prism', () => ({
  oneDark: {},
}))
jest.mock('components/Button', () => ({
  Button: ({ onClick }: any) => <button onClick={onClick}>Button</button>,
}))

describe('<CodeDisplay />', () => {
  test('renders correctly', () => {
    expect.assertions(1)

    const { asFragment } = renderWithProviders(<CodeDisplay code='' />)

    const buttonShowCode = screen.getAllByRole('button')[1]
    fireEvent.click(buttonShowCode)

    expect(asFragment()).toMatchSnapshot()
  })
})
