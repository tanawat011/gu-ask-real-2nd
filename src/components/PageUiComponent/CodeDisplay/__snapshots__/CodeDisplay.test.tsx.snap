// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeDisplay /> renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  border-bottom-right-radius: 0.5rem;
  border-bottom-left-radius: 0.5rem;
  border-top-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgb(75 85 99 / var(--tw-border-opacity));
  --tw-bg-opacity: 1;
  background-color: rgb(55 65 81 / var(--tw-bg-opacity));
  padding-left: 1.25rem;
  padding-right: 1.25rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  display: grid;
  grid-auto-flow: column;
  gap: 0.25rem;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <button
          data-testid="button"
        >
          Button
        </button>
        <button
          data-testid="button"
        >
          Button
        </button>
      </div>
    </div>
    <div>
      SyntaxHighlighter
    </div>
  </div>
</DocumentFragment>
`;
